net/SessionManager.c

```diff
@@ -491,7 +493,7 @@ static void checkTimedOutSessions(struct SessionManager_pvt* sm)
         int64_t now = Time_currentTimeMilliseconds(sm->eventBase);
 
         // Check if the session is timed out...
-        if (now - sess->pub.timeOfLastIn > sm->pub.sessionTimeoutMilliseconds) {
```

Comment:
IMO KeepAlive is just additional message for keeping the session alive. It doesn't have to be keep alive message any message should keep the session alive. KeepAlive messages just make sure that it is kept alive (but they may be dropped (UDP).